4/README.md

```diff
@@ -0,0 +1,346 @@
+  (one of *READ*, *WRITE*, *READ_WRITE*, *READ_WITH_WRITE*).
+* *display level*, an enumeration describing visibility level of the Attribute
+  (one of *EXPERT*, *OPERATOR*).
+
+> **Note:**
+> Although it is possible to use a wide range of characters in the Attribute's
+> name, it is RECOMMENDED to use only numbers, letters (upper- and lower-case)
+> and an underscore (`_`) to ensure compatibility with various tools and client
+> applications. Also note that the Attribute name SHOULD contain at least one
+> letter and SHOULD NOT start with a digit.
+
+An Attribute can be read from, if *writable* is one of *READ*, *READ_WRITE*,
+*READ_WITH_WRITE*.
+
+An Attribute can be written to, if *writable* is one of *WRITE*, *READ_WRITE*.
+
+[^1]: In current implementation it is possible to define more than one
+Attribute with the same name, but only one can be accessed externally.
+
+If *data format* is *SPECTRUM*,
+an Attribute MUST have associated following additional static metadata:
+* *max dim x*, an integer describing the maximum allowed number of data
+  elements.
+  The Attribute MUST be able to store up to *max dim x* data elements.
+  It MUST be greater than 0.
+
+If *data format* is *IMAGE*,
+an Attribute MUST have associated following additional static metadata:
+* *max dim x*, an integer describing the maximum allowed number of data
+  elements in dimension X.
+  The Attribute MUST be able to store up to *max dim x* data elements
+  in the X dimension.
+  It MUST be greater than 0,
+* *max dim y*, an integer describing the maximum allowed number of data
+  elements in dimension Y.
+  The Attribute MUST be able to store up to *max dim y* data elements
+  in the Y dimension.
+  It MUST be greater than 0.
+
+If *writable* is *READ_WITH_WRITE*,
+an Attribute MUST have associated following additional static metadata:
+* *writable attribute name*, a string describing the *name* of associated
+  writable attribute.
+  It MUST point to an attribute.
+  The Attribute pointed to MUST have *writable* set to one of *WRITE*,
+  *READ_WRITE*.
+
+If *data type* is *ENUM*,
+an Attribute MAY have associated following additional static metadata:
+* *enum labels*, a list of strings describing textual representation
+  of enumerated values.
+  Its elements MAY contain any character (including multi-byte characters)
+  except that they MUST NOT contain `\0` character.
+  It MAY be empty.
+
+
+#### Memorized attribute
+
+An Attribute MUST have associated following static metadata:
+* *memorized*, a flag describing whether *set value* is stored in the database
+  and restored during Attribute initialization,
+* *write hardware at init*, a flag describing whether *set value* is
+  written to the Attribute during initialization or *init* call.
+  It is effective only if *memorized* is also set.
+
+To set *memorized*, following constraints MUST be satisfied:
+* *data format* MUST be *SCALAR*,
+* *data type* MUST NOT be *STATE* or *ENCODED*,
```

Comment:
So, do you propose to remove this constrains from the RFC? Or shall I add a note here that this will be changed in the future?

Reply:
> 
> 
> So, do you propose to remove this constrains from the RFC? Or shall I add a note here that this will be changed in the future?

This constraint on the type (STATE or ENCODED) might not exist in the latest version of JTango.

JTango does not (no longer) support(s) spectrum or image memorized attributes mainly due to performance issues:

https://github.com/tango-controls/JTango/blob/jtango-9-lts/server/src/main/java/org/tango/server/attribute/AttributeImpl.java#L482-L492

If an implementation manages to support these kinds of memorized attributes in an efficient manner, I would not forbid it.

We could add a note about what kind of memorized attributes are currently supported by JTango and cppTango.

In any case, what is important here is the concept of memorized attribute.